Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -479,29 +447,71 @@ namespace AZ::Reflection
-                    // the group and its parent when its parent would be closed.
-                    if (nodeData.m_entryClosed)
+                    EndNode();
+                }
+                // Iterate over non serialized elements to see if any of them should be added
+                const char* elementName = "";
+                if (nodeData.m_classElement&& nodeData.m_classElement->m_editData&& nodeData.m_classElement->m_editData->m_name)
+                {
+                    elementName = nodeData.m_classElement->m_editData->m_name;
+                }
+                auto iter = m_nonSerializedElements.begin();
+                while (iter != m_nonSerializedElements.end())
+                {
+                    // If the parent of the element that was just created has the same name as the parent of any non serialized elements,
+                    // and the element that was just created is the element immediately before any non serialized element, create that serialized element
+                    if (m_stack.back().m_classData && m_stack.back().m_classData->m_name == iter->second.m_classData->m_name &&
+                        elementName == iter->first)
                     {
-                        m_visitor->VisitObjectEnd(*this, *this);
-                        m_stack.pop_back();
-                        EndNode();
+                        if (iter->second.m_classElement->m_editData->m_elementId == AZ::Edit::ClassElements::Group)
+                        {
+                            AZStd::string_view groupName = iter->second.m_classElement->m_editData->m_description;
+                            if (!m_stack.back().m_group.empty())
+                            {
+                                // If we're superceding a previous group, finish visiting the previous group first
+                                m_visitor->VisitObjectEnd(*this, *this);
+                                m_stack.pop_back();
+                            }
+                            if (!groupName.empty())
+                            {
+                                // If our group has a name, we should actually call VisitObjectEnd, so continue with the group set
+                                m_stack.push_back(iter->second);
+                                m_stack.back().m_group = groupName;
+                                CacheAttributes();
+                                m_visitor->VisitObjectBegin(*this, *this);
+                            }
+                            //else //there is no group to visit?
+                            //{
+                            //    m_stack.back().m_entryClosed = true;
+                            //    m_visitor->VisitObjectEnd(*this, *this);
+                            //    m_stack.pop_back();
+                            //    EndNode();
+                            //    return true;
+                            //}
```

Comment:
Thanks, that was originally written by NVS, I had it commented while I investigated whether we needed to add this case, but since we never add a group to the stack if it doesn't have a group description then this case is not needed anymore